refactor #673

adamralph opened the pull request.

blairconrad commented:
Generally LGTM, @adamralph. Thanks. One question, though. When I used going strings, I got a compiler warning. You have no such problem?

adamralph commented:
@blairconrad not sure what you mean by "going strings" but if you mean string interpolation, I guess it's because we have [CA1303](https://msdn.microsoft.com/en-us/library/ms182187.aspx) switched off for our test projects?

blairconrad commented:
Sorry, "bling" strings (string interpolation). Didn't notice the autocorrect.
Maybe turning off CA1303 does help. I my have gotten the warning at the point I was creating the exception messages and overcompensated.

blairconrad commented:
"may have gotten"

adamralph commented:
We have CA1303 enabled for our production projects so I guess you would have gotten the warning there.

thomaslevesque commented on tests/FakeItEasy.Specs/ConfiguringPropertySetterSpecs.cs:
```diff
@@ -33,9 +32,11 @@ public interface IHaveInterestingProperties
+            "Then an argument null exception is thrown"
+                .x(() => exception.Should().BeAnExceptionOfType<ArgumentNullException>());
+
+            "And the parameter name should be 'propertySpecification'"
```
Didn't we decide earlier that step descriptions shouldn't contain "should" ? `"And the parameter name is 'propertySpecification'"`

adamralph replied:
Yep, agreed. Good spot. I'll amend a bit later
On 14 May 2016 12:46, "Thomas Levesque" <notifications@github.com> wrote:

> In tests/FakeItEasy.Specs/ConfiguringPropertySetterSpecs.cs
> <https://github.com/FakeItEasy/FakeItEasy/pull/704#discussion_r63275041>:
>
> > @@ -33,9 +32,11 @@ public interface IHaveInterestingProperties
> >              "When assignment of a property is configured using a null expression"
> >                  .x(() => exception = Record.Exception(() => A.CallToSet<int>(null)));
> >
> > -            "Then the call to configure throws"
> > -                .x(() => exception.Should().BeAnExceptionOfType<ArgumentNullException>()
> > -                    .And.ParamName.Should().Be("propertySpecification"));
> > +            "Then an argument null exception is thrown"
> > +                .x(() => exception.Should().BeAnExceptionOfType<ArgumentNullException>());
> > +
> > +            "And the parameter name should be 'propertySpecification'"
>
> Didn't we decide earlier that step descriptions shouldn't contain "should"
> ? "And the parameter name is 'propertySpecification'"
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/FakeItEasy/FakeItEasy/pull/704/files/80022e82dab8d9ccd9fedc96db1ce60801872b75#r63275041>
>

thomaslevesque commented:
Looks good to me, except:
- a minor comment on the wording in step descriptions
- the PR description isn't really accurate; it's mostly a refactoring of the specs, the modification of the XML docs is only a small part of the changes

adamralph commented:
@thomaslevesque thanks for reviewing. Please see the extra commit I've added in the middle. If you're happy, I'll squash and force push so you can merge.

I've also split out the XML tweaks into #709

thomaslevesque commented:
Perfect, thanks :+1:

adamralph commented:
Squashed.

thomaslevesque commented:
Just wondering... do we really need a different build job for approval tests?

adamralph commented:
@thomaslevesque I explained my reasoning for not including the approval test in the default rake tasks and the main build in https://github.com/FakeItEasy/FakeItEasy/issues/701.

I believe doing so would be harmful to our contributor ecosystem.

thomaslevesque commented:
Ah, yes, I remember now... thanks!
